snakemake/__init__.py

```diff
@@ -633,18 +638,27 @@ def snakemake(
             scheduler_solver_path=scheduler_solver_path,
             conda_base_path=conda_base_path,
             check_envvars=not lint,  # for linting, we do not need to check whether requested envvars exist
+            benchmark_output=benchmark_output,
+            benchmark_all=benchmark_all,
             all_temp=all_temp,
             local_groupid=local_groupid,
             keep_metadata=keep_metadata,
             latency_wait=latency_wait,
         )
         success = True
-
         workflow.include(
             snakefile,
             overwrite_default_target=True,
             print_compilation=print_compilation,
         )
+        if benchmark_all is not None:
+            if not forceall:
+                logger.warning(
+                    "Warning: Benchmarking all rules but --forceall is not set. Resulting benchmarks maybe incomplete or outdated!"
```

Comment:
I think this warning is unnecessary

Reply:
How so? If a rule is skipped due to a previous run, but if hardware (eg. cloud platforms) have changed since then, current run would still grab metrics from previous files and thus they would be inaccurate, no?